packages/@dcos/sorting/__tests__/compare-as-number-test.ts

```diff
@@ -0,0 +1,34 @@
+import { compareAsNumber, ComparisonResult } from "../";
+
+const idLense = (x: any) => x;
+
+describe("compareAsNumber", () => {
+  it("returns 0 if any on the values is null", () => {
```

Comment:
What about making `null` and `undefined` be 0 or Inf while sorting numbers?

The current solution does not seem like it works for me.
```js
[1,null,2, null, 2, 1].sort((a,b) => a == null || b == null ? 0 : a - b)
> [ 1, null, 2, null, 1, 2 ]
```

*short hypothesis*
The reason would be null being equivalent to any other value they never move.

*long hypothesis*

making `1 > null be 0`, and `1 < null also be 0` breaks antisymmetry `if a ≤ b and b ≤ a, then a = b` which makes our comparators not being a Partial Order [0].

Failing to be a partial order means that `null` will be equal to every value, making null never sort. 

[0] - https://en.wikipedia.org/wiki/Partially_ordered_set

Reply:
I said something wrong. We actually need a Total Order for sorting (https://en.wikipedia.org/wiki/Total_order). Total Order also needs to respect `asymmetry`.

Reply:
I don't think it breaks antisymmetry, to be honest. the value zero expresses equality, so it's more like everything is equal to null, which I think is okay.

Reply:
Maybe I should check with product what should be the case. the null behavior is what we currently have, but that, of course, does not mean we need to keep it :)

I will follow up on this and inform you ;)

Reply:
ok.